Assets/Dev/Si Wai/PermanentPlayerStatistic.cs

```diff
@@ -0,0 +1,106 @@
+
+using UnityEngine;
+[System.Serializable]
+public class PermanentPlayerStatistic : SaveSystem
+{
+    public PermanentStatistic Speed = new();
+    public PermanentStatistic JumpForce = new();
+    public PermanentStatistic MaxHealth = new();
+    private float health;
+    public float Health {
+        get => health;
+        set => health = value > 0 ? value : 0;
+    }
+    private int crumbs;
+    public int Crumbs
+    {
+        get => crumbs;
+        set => crumbs = value > 0 ? value : 0;
+    }
+    public PermanentStatistic AttackSpeedMultiplier = new();
+    public PermanentStatistic AttackDamageMultiplier = new();
+    public PermanentStatistic DodgeCooldown = new();
+    public PermanentStatistic DoubleJumpsCount = new();
+    protected override string Prefix => "permanentPlayerStatistic";
+
+    public override void Init() {
+        // using json strings to save the stats
+        string json;
+
+        json = JsonUtility.ToJson(Speed.ListWithModifications());
+        Add("speed", json);
+
```

Comment:
clean code preference, sort this into a dictionary and loop over it 

 
    private Dictionary<string, PermanentStatistic> statistics = new();

  public override void Init()
    { 
        statistics["speed"] = Speed;
        statistics["jumpForce"] = JumpForce;
        statistics["maxHealth"] = MaxHealth;
        statistics["attackSpeedMultiplier"] = AttackSpeedMultiplier;
        statistics["attackDamageMultiplier"] = AttackDamageMultiplier;
        statistics["dodgeCooldown"] = DodgeCooldown;
        statistics["doubleJumpsCount"] = DoubleJumpsCount;

        foreach (var stat in statistics)
        {
            Add(stat.Key, JsonUtility.ToJson(stat.Value.ListWithModifications()));
        }

Reply:
agreed